scan/action.yml

```
@@ -104,9 +96,22 @@ runs:
           echo "::warning:: No SARIF file generated from scorecard scan"
         fi
 
-    - name: Upload trivy SARIF file
+    - name: Upload SARIF files
       uses: github/codeql-action/upload-sarif@ce729e4d353d580e6cacd6a8cf2921b72e5e310a
-      if: success() || failure()
+      if: (success() || failure()) && github.event.repository.private == false
       continue-on-error: true
       with:
         sarif_file: scan_reports
+
+    # Upload SARIF artifacts for private repositories
+    - name: Upload Trivy SARIF artifacts
+      if: (success() || failure()) && github.event.repository.private == true
+      uses: actions/upload-artifact@v4
+      with:
+        name: sarif_reports
+        path: scan_reports
+
+    - name: Fail on severity
+      shell: bash
+      if: inputs.fail_severity != ''
+      run: ./scripts/fail-on-severity.sh "${{ inputs.fail_severity }}"
```

Script path - action.yml calls ./scripts/fail-on-severity.sh but the script is at scan/scripts/. May need the following depending on working directory?

```suggestion
      run: ./scripts/fail-on-severity.sh "${{ github.action_path }}/scripts/fail-on-severity.sh"
```